samples/bidirectional-chat/csharp/Function.cs

```diff
@@ -12,25 +12,23 @@ namespace FunctionApp
 {
     public class SimpleChat : ServerlessHub
     {
-        private const string Hub = nameof(SimpleChat);
         private const string NewMessageTarget = "newMessage";
         private const string NewConnectionTarget = "newConnection";
 
         [FunctionName("negotiate")]
-        public static SignalRConnectionInfo GetSignalRInfo(
-            [HttpTrigger(AuthorizationLevel.Anonymous)] HttpRequest req,
-            [SignalRConnectionInfo(HubName = Hub, UserId = "{headers.x-ms-signalr-user-id}")] SignalRConnectionInfo connectionInfo)
+        public SignalRConnectionInfo Negotiate([HttpTrigger(AuthorizationLevel.Anonymous)]HttpRequest req)
         {
-            return connectionInfo;
+            return Negotiate(req.Headers["x-ms-signalr-user-id"], GetClaims(req.Headers["Authorization"]));
         }
 
-        [FunctionName(nameof(Connected))]
-        public async Task Connected([SignalRTrigger]InvocationContext invocationContext, ILogger logger)
+        [FunctionName(nameof(OnConnected))]
+        public async Task OnConnected([SignalRTrigger]InvocationContext invocationContext, ILogger logger)
```

Comment:
How add `OnConnected` and `OnDisconnected` as virtual methods in base class?

Reply:
We don't know parameters of these two methods. Just like `Configure` and `ConfigureService` in `Startup`, which can only restrict by convention.